pkg/apis/meta/v1/types.go

```diff
@@ -91,6 +91,14 @@ type OwnerReference struct {
 	// If true, this reference points to the managing controller.
 	// +optional
 	Controller *bool `json:"controller,omitempty" protobuf:"varint,6,opt,name=controller"`
+	// If true, AND if the owner has the "DeletingDependents" finalizer, then
+	// the owner cannot be deleted from the key-value store until this
+	// reference is removed.
+	// Defaults to false.
+	// To set this field, a user needs "delete" permission of the owner,
+	// otherwise 422 (Unprocessable Entity) will be returned.
+	// +optional
+	BlockOwnerDeletion *bool `json:"blockOwnerDeletion,omitempty" protobuf:"varint,7,opt,name=blockOwnerDeletion"`
```

Comment:
It is a field in OwnerReference, which is a type used in metadata.

We are adding Foreground Garbage Collection, meaning the owner object will not be deleted until all its dependents are garbage collected. The side effect is that a user who can update the dependent object has the power to hold the deletion of the owner, we think this breaks abac. As a solution, we add the `HoldOwnerDeletion` field to OwnerReference, only the ownerReferences with `HoldOwnerDeletion=true` will be counted during the foreground GC. Only the user who has the update && delete permission of the owner can set this field to true.